src/SpoutReader.php

```diff
@@ -0,0 +1,245 @@
+
+        $activeSheet = null === $activeSheet ? 0 : (int) $activeSheet;
+        foreach ($reader->getSheetIterator() as $sheet) {
+            if ($sheet->getIndex() === $activeSheet) {
+                break;
+            }
+        }
+
+        if (!$reader->getSheetIterator()->valid()) {
+            throw new ReaderException(sprintf('Sheet at index %d is not found', $activeSheet));
+        }
+        $this->sheet = $reader->getSheetIterator()->current();
+
+        if (null !== $headerRowNumber) {
+            $this->setHeaderRowNumber($headerRowNumber);
+        }
+    }
+
+    /**
+     * @param \SplFileObject $file
+     * @param $shouldPreserveEmptyRows
+     *
+     * @return \Box\Spout\Reader\XLSX\Reader
+     */
+    private function createReaderForFile(\SplFileObject $file, $shouldPreserveEmptyRows)
+    {
+        /** @var \Box\Spout\Reader\XLSX\Reader $reader */
+        $reader = ReaderFactory::create(Type::XLSX);
+        $reader->setShouldPreserveEmptyRows($shouldPreserveEmptyRows);
+        $reader->open($file->getPathname());
+
+        return $reader;
+    }
+
+    /**
+     * Return the current row as an array
+     *
+     * If a header row has been set, an associative array will be returned
+     *
+     * @return array
+     */
+    public function current()
+    {
+        $row = $this->sheet->getRowIterator()->current();
+
+        // If the CSV has column headers, use them to construct an associative
+        // array for the columns in this line
+        if (!empty($this->columnHeaders)) {
+            // Count the number of elements in both: they must be equal.
+            // If not, ignore the row
+            if (count($this->columnHeaders) === count($row)) {
+                return array_combine(array_values($this->columnHeaders), $row);
+            }
+        } else {
+            // Else just return the column values
+            return $row;
+        }
+    }
+
+    /**
+     * Get column headers
+     *
+     * @return array
+     */
+    public function getColumnHeaders()
+    {
+        return $this->columnHeaders;
+    }
+
+    /**
+     * Set column headers
+     *
+     * @param array $columnHeaders
+     */
+    public function setColumnHeaders(array $columnHeaders)
+    {
+        $this->columnHeaders = $columnHeaders;
+    }
+
+    /**
+     * Rewind the file pointer
+     *
+     * If a header row has been set, the pointer is set just below the header
+     * row. That way, when you iterate over the rows, that header row is
+     * skipped.
+     */
+    public function rewind()
+    {
+        $this->sheet->getRowIterator()->rewind();
+        if (null !== $this->headerRowNumber) {
+            $this->seekIndex($this->headerRowNumber + 2);
+        }
+    }
+
+    /**
+     * Set header row number
+     *
+     * @param integer $rowNumber Number of the row that contains column header names
+     */
+    public function setHeaderRowNumber($rowNumber)
+    {
+        $rowNumber = (int) $rowNumber;
```

Comment:
I would rather use a static type hint for this method.

Reply:
Need to require PHP7